shipstation_integration/orders.py

```diff
@@ -94,8 +94,19 @@ def validate_order(
 	if not order:
 		return False
 
-	# if an order already exists, skip
-	if frappe.db.get_value("Sales Order", {"shipstation_order_id": order.order_id, "docstatus": 1}):
+	# if an order already exists, skip, unless the status needs to be updated
+	existing_order = frappe.db.get_value(
+		"Sales Order", {"shipstation_order_id": order.order_id}, ["name", "status"], as_dict=True
+	)
+	if existing_order:
+		new_status, new_docstatus = get_erpnext_status(order.order_status)
+		if existing_order.status != new_status:
+			frappe.db.set_value(
+				"Sales Order",
+				existing_order.name,
+				{"status": new_status, "docstatus": new_docstatus},
+				update_modified=False,
+			)
```

Comment:
@batonac I'm not sure explicitly setting the status and docstatus would work, since we'd probably skip through a bunch of other controller functions. Also, I think ideally we should be letting Frappe decide what status a document should have, and run the functions that generate the status effect.

Maybe we want to model these specific updates using [webhooks](https://help.shipstation.com/hc/en-us/articles/360025856252-ShipStation-Webhooks)? Or use the ORM (`doc.submit()`, `doc.cancel()`, etc.) to let the controller handle it?

CC: @agritheory

Reply:
Good point! I wasn't completely happy with the separate handling of docstatus in create_erpnext_order and validate_order in this PR, but it wasn't immediately apparent to me how to do that yesterday. I think I have a way forward, and will update this PR shortly.

Reply:
@Alchez, I've factored around this concern, and I believe it represents a sizable improvement, especially since the pre/post submit hooks are integrated on every submit. You might not like my passing of the `so` as a `dict` in `validate_order`, but I think that's the most performant solution, even if not the clearest.

I'd be grateful if y'all could take this from here, if you'd prefer to adjust any of this.